src/compiler/core.ts

```diff
@@ -429,7 +433,7 @@ namespace ts {
                         addRange(result, v);
                     }
                     else {
-                        result.push(v);
+                        result.push(v as T);

```

Comment:
For some reason I thought changing `isArray` would break things. But it seems to work fine.